src/v2/providers/eventarc.ts

```diff
@@ -0,0 +1,163 @@
+  /**
+   * ID of the channel. Can be either:
+   *   * fully qualified channel resource name:
+   *     `projects/{project}/locations/{location}/channels/{channel-id}`
+   *   * partial resource name with location and channel id, in which case
+   *     the runtime project ID of the function will be used:
+   *     `locations/{location}/channels/{channel-id}`
+   *   * partial channel-id, in which case the runtime project ID of the
+   *     function and `us-central1` as location will be used:
+   *     `{channel-id}`
+   * 
+   * If not specified, the default Firebase channel will be used:
+   * `projects/{project}/locations/us-central1/channels/firebase`
+   */
+   channel?: string;
+
+  /**
+   * Eventarc event exact match filter.
+   */
+   filters?: Record<string, string>;
+ }
+
+export type CloudEventHandler = (event: CloudEvent<any>) => any | Promise<any>;
+
+/** Handle an Eventarc event published on the default channel. */
+export function onCustomEventPublished<T = any>(
+  eventType: string,
+  handler: CloudEventHandler
+): CloudFunction<CloudEvent<T>>;
+
+export function onCustomEventPublished<T = any>(
+  eventType: string,
+  opts: EventarcTriggerOptions,
+  handler: CloudEventHandler
+): CloudFunction<CloudEvent<T>>;
+
+export function onCustomEventPublished<T = any>(
+  eventType: string,
+  optsOrHandler:
+    CloudEventHandler
+    | EventarcTriggerOptions,
+  handler?: CloudEventHandler,
+): CloudFunction<CloudEvent<T>> {
+  if (arguments.length < 2) {
+    throw new Error(
+      'Missing required parameters.'
+    );  
+  }
+  if (arguments.length == 2) {
+    if (typeof optsOrHandler !== 'function') { 
+      throw new Error(
+        'Expected second parameter to be the event handler function.'
+      );
+    }
+    handler = optsOrHandler as CloudEventHandler;
+    optsOrHandler = {} as EventarcTriggerOptions;
+  } else if (arguments.length == 3) {
+    optsOrHandler = optsOrHandler as EventarcTriggerOptions;
+  }
+
+  const opts = optsOrHandler as EventarcTriggerOptions;
+  const func = (raw: CloudEvent<unknown>) => {
+    return handler(raw as CloudEvent<T>);
+  };
+
+  func.run = handler;
+
+  const channel = opts.channel ?? 'locations/us-central1/channels/firebase';
+
+  Object.defineProperty(func, '__trigger', {
+    get: () => {
+      const baseOpts = options.optionsToTriggerAnnotations(
+        options.getGlobalOptions()
+      );
+      const specificOpts = options.optionsToTriggerAnnotations(opts);
+      const trigger = {
+        platform: 'gcfv2',
+        ...baseOpts,
+        ...specificOpts,
+        labels: {
+          ...baseOpts?.labels,
+          ...specificOpts?.labels,
+        },
+        eventTrigger: {
+          eventType,
+          filters: opts.filters,
+          channel,
+        },
+      };
+      if (!opts.filters) {
+        delete trigger.eventTrigger.filters;
+      }
+
+      return trigger;
+    },
+  });
```

Comment:
I think we have a soft rule around "we won't be adding `__trigger` annotation to new trigger types. Feel free to remove this.

Reply:
Today `firebase deploy` (parseTriggers) uses this, I can't make it work without adding this...

Reply:
Figured it out. Done.